client/mm/libxc/interface.go

```diff
@@ -0,0 +1,92 @@
+package libxc
+
+import (
+	"context"
+	"fmt"
+
+	"decred.org/dcrdex/dex"
+)
+
+// ExchangeBalance holds the available and locked balances of an asset
+// on a CEX.
+type ExchangeBalance struct {
+	Available uint64 `json:"available"`
+	Locked    uint64 `json:"locked"`
+}
+
+// TradeUpdate is a notification sent when the status of a trade on the CEX
+// has been updated.
+type TradeUpdate struct {
+	TradeID  string
+	Complete bool // cancelled or filled
+}
+
+// Market is the base and quote assets of a market on a CEX.
+type Market struct {
+	Base  uint32 `json:"base"`
+	Quote uint32 `json:"quote"`
+}
+
+// CEX implements a set of functions that can be used to interact with a
+// centralized exchange's spot trading API. All rates and quantities
+// when interacting with the CEX interface will adhere to the standard
+// rates and quantities of the DEX.
+type CEX interface {
+	Connect(context.Context) error
+	// Balance returns the balance of an asset at the CEX.
+	Balance(symbol string) (*ExchangeBalance, error)
+	// Balances returns a list of all asset balances at the CEX. Only assets that are
+	// registered in the DEX client will be returned.
+	Balances() (map[uint32]*ExchangeBalance, error)
+	// CancelTrade cancels a trade on the CEX.
+	CancelTrade(baseSymbol, quoteSymbol, tradeID string) error
+	// GenerateTradeID returns a trade ID that must be passed as an argument
+	// when calling Trade. This ID will be used to identify updates to the
+	// trade. It is necessary to pre-generate this because updates to the
+	// trade may arrive before the Trade function returns.
+	GenerateTradeID() string
+	// Markets returns the list of markets at the CEX.
+	Markets() ([]*Market, error)
+	// SubscribeCEXUpdates returns a channel which sends an empty struct when
+	// the balance of an asset on the CEX has been updated.
+	SubscribeCEXUpdates() <-chan interface{}
+	// SubscribeMarket subscribes to order book updates on a market. This must
+	// be called before calling VWAP.
+	SubscribeMarket(baseSymbol, quoteSymbol string) error
+	// SubscribeTradeUpdates returns a channel that the caller can use to
+	// listen for updates to a trade's status. If the integer returned from
+	// this function is passed as the updaterID argument to Trade, then updates
```

Comment:
Says if but the updaterID appears to be required in Trade.

Why do we need two nonce? I feel like this should be doable with one... The `bnc.tradeUpdaters` could be keyed with the same string, or something.

Reply:
The `updaterID` identifies which channel to send the trade updates on, and `tradeID` identifies the trade itself. Ideally, `Trade` would return the `tradeID` and it would not have to be passed in, but Binance may send a trade update over websocket even before the Trade http response has returned. Therefore, the caller of `Trade` needs to know the `tradeID` before they call `Trade`.